pkg/cloud/services/compute/instance.go

```diff
@@ -610,16 +609,29 @@ func (s *Service) getImageID(imageName string) (string, error) {
 	}
 }
 
-func (s *Service) AssociateFloatingIP(instanceID, floatingIP string) error {
-	opts := floatingips.AssociateOpts{
-		FloatingIP: floatingIP,
+func (s *Service) GetControlPlanePort(instance *infrav1.Instance) (*ports.Port, error) {
+	mc := metrics.NewMetricPrometheusContext("port", "list")
+	portOpts := ports.ListOpts{
+		DeviceID: instance.ID,
+		FixedIPs: []ports.FixedIPOpts{
+			{
+				IPAddress: instance.IP,
+			},
+		},
+		Limit: 1,
 	}
-	mc := metrics.NewMetricPrometheusContext("floating_ip", "update")
-	err := floatingips.AssociateInstance(s.computeClient, instanceID, opts).ExtractErr()
+	allPages, err := ports.List(s.networkClient, portOpts).AllPages()
 	if mc.ObserveRequest(err) != nil {
-		return err
+		return nil, fmt.Errorf("lookup control plane port for server %s: %w", instance.ID, err)
```

Comment:
lookup port for server %s: %w is better because this is used for port of bastion also.